### DISCUSSION TITLE
Pexp_newtype vs. Pexp_function (in OCaml 5.2.0) ?

### DISCUSSION THREAD

**Discussion by chetmurthy** [ocaml/ocaml#13122] [Q&A]
In OCaml 5.1.0, there were three different constructors for functions:
1. Pexp_function -- for "function p1 -> e1 | p2 -> e2 | etc"
2. Pexp_fun -- for "fun p1 -> e1"
3. Pexp_newtype -- for "fun (type t) -> e"

With OCaml 5.2.0, all three are unified into Pexp_function, but Pexp_newtype still exists.  Is there a reason for this I'm not seeing?  The specific reason I ask, is that if you're doing something like `ppx_metaquot`, the fact that there are two different ASTs for  what is actually the same surface syntax -- two entirely different AST constructors -- would seem to make it difficult to implement a metaquot facility.

**Comment by gasche** [comment]
This is a reasonable question, and I suspect that this was not really asked explicitly during the work on the unified `Pexp_function` construction in #12236. (Or maybe there was the idea to change as little as possible.) We could still proceed to remove `Pexp_newtype` now.
